packages/shared/src/settings/background.ts

```diff
@@ -17,17 +17,22 @@ type BackgroundSettingsType = {
     id?: string;
     defaultValue?: boolean;
     defaultColor?: Color;
+    hasDefaultColor?: boolean;
+    switchLabel?: string;
 };
 
 export const getBackgroundSettings = (options?: BackgroundSettingsType): SettingBlock => {
+    const showDefaultColor = options?.hasDefaultColor !== undefined ? options?.hasDefaultColor : true;
     const hasId = options?.id ? `hasBackground${options.id}` : 'hasBackground';
     const colorId = options?.id ? `backgroundColor${options.id}` : 'backgroundColor';
-    const defaultColor = options?.defaultColor || BACKGROUND_COLOR_DEFAULT_VALUE;
+    const defaultColor = showDefaultColor ? options?.defaultColor || BACKGROUND_COLOR_DEFAULT_VALUE : undefined;
```

Comment:
Quick question. Why is the extra prop `hasDefaultColor` needed? Can't we just check to see if the `defaultColor` prop has been supplied?

Reply:
the idea is, that you don't have to put a defaultColor. As atm we use everywhere the BACKGROUND_COLOR_DEFAULT_VALUE therefore we don't need to pass it each time. You know what I mean? I can also adapt it like you said. Wdyt?